feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go

```diff
@@ -395,40 +395,36 @@ func TestComponentParent(t *testing.T) {
 
 	for _, tc := range cases {
 		t.Run(tc.desc, func(t *testing.T) {
-			t.Logf("Found card list for Component Type %v : %v", tc.componentType, cardList[tc.desc])
+			t.Logf("Found Component list for Type %v : %v", tc.componentType, cardList[tc.desc])
 			if len(cardList[tc.desc]) == 0 {
-				t.Fatalf("Get Card list for %q: got 0, want > 0", dut.Model())
+				t.Fatalf("Get Component list for %q: got 0, want > 0", dut.Model())
 			}
 			// Validate parent component
 			for _, card := range cardList[tc.desc] {
-				t.Logf("Validate card %s", card)
+				t.Logf("Validate component %s", card)
 				cardName := card
+				foundChassis := false
 				for {
 					parent := gnmi.Lookup(t, dut, gnmi.OC().Component(cardName).Parent().State())
 					val, present := parent.Val()
 					if !present {
 						t.Fatalf("Parent not present for %q: got %v, want true", card, parent.IsPresent())
 					} else {
 						got := gnmi.Get(t, dut, gnmi.OC().Component(val).Type().State())
-						if tc.desc == "SwitchChip" || tc.desc == "FabricChip" {
-							if got == componentParent["FabricChip"] || got == componentParent["SwitchChip"] {
-								t.Logf("Got expected parent for card %s", card)
-								break
-							}
-						} else {
-							if got == tc.parent {
-								t.Logf("Got expected parent for card %s", card)
-								break
-							}
+						if got == chassisType {
+							foundChassis = true
+							t.Logf("Found chassis component in the hierarchy tree of component %s", card)
+							break
```

Comment:
When we get out of the inner 'for' loop, we need to know if the loop was exited due to the 'break' ('pass' scenario) or because we reached the top of the component tree and didn't find what we were looking for ('fail' scenario), so that we could log the appropriate 'fail' message. Had used the foundChassis flag for this purpose.

Reply:
I misread and forgot this validation is happening for every component in the list, not just one. Just leave as is.